server/https_proxy.go

```diff
@@ -0,0 +1,75 @@
+package server
+
+import (
+	"crypto/tls"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
```

Comment:
Can you document how this works?  I'm not sure what's going on here.  For example, why is the source closed on every transfer?  Wouldn't that break the connection?

Reply:
io.Copy copies data until EOF is read from the source, or an error occurs.  So the source is only closed after all of the data is copied.

Reply:
Added a comment in the code.